src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1613,6 +1503,28 @@ export class NetworkAreaDiagramViewer {
         }
     }
 
+    private getZoomLevel(maxDisplayedSize: number): number {
+        if (maxDisplayedSize >= 20000) {
```

Comment:
Could you use an array instead of these separated values?
Wouldn't it make sense to have that array as input (with a default hardcoded array)? instead of an hardcoded array.